logstash-core/lib/logstash/runner.rb

```diff
@@ -241,9 +241,11 @@ def execute
     end
 
     if setting("config.test_and_exit")
-      config_loader = LogStash::Config::Loader.new(logger)
-      config_str = config_loader.format_config(setting("path.config"), setting("config.string"))
       begin
+        config_loader = LogStash::Config::SOURCE_LOADER.create(@settings)
+
+        # TODO(ph): multiple pipeline will affect this
+        config_str = config_loader.pipeline_config.config_string
```

Comment:
@jsvd and myself discussed on zoom.

We cannot easily change the signature of the Pipeline because some plugins tests use it directly.
But we can create another class maybe `RubyPipeline` for lack of a better name that use the new signature and the old pipeline will change the internal.